File: server/src/main/java/com/grpc/empuje_comunitario/infrastructure/persistence/UserEntity.java
package com.grpc.empuje_comunitario.infrastructure.persistence;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "users")
public class UserEntity {

    @Id
    private String id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String lastname;

    private String phone;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String role;

    @Column(nullable = false)
    private boolean active;

    public UserEntity() {
    }

    public UserEntity(String id, String username, String name, String lastname,
                      String phone, String email, String password, String role, boolean active) {
        this.id = id;
        this.username = username;
        this.name = name;
        this.lastname = lastname;
        this.phone = phone;
        this.email = email;
        this.password = password;
        this.role = role;
        this.active = active;
    }

    public String getId() { return id; }
    public String getUsername() { return username; }
    public String getName() { return name; }
    public String getLastname() { return lastname; }
    public String getPhone() { return phone; }
    public String getEmail() { return email; }
    public String getPassword() { return password; }
    public String getRole() { return role; }
    public boolean isActive() { return active; }
}

File: server/src/main/java/com/grpc/empuje_comunitario/infrastructure/persistence/UserEntity.kt
package com.grpc.empuje_comunitario.infrastructure.persistence

import com.grpc.empuje_comunitario.domain.user.User
import com.grpc.empuje_comunitario.domain.user.asString
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "users")
data class UserEntity(
    @Id
    val id: String = "",

    @Column(unique = true, nullable = false)
    val username: String = "",

    @Column(nullable = false)
    val name: String = "",

    @Column(nullable = false)
    val lastname: String = "",

    val phone: String? = null,

    @Column(unique = true, nullable = false)
    val email: String = "",

    @Column(nullable = false)
    val password: String = "",

    @Column(nullable = false)
    val role: String = "",

    @Column(nullable = false)
    val active: Boolean = false
) {
    // No-arg constructor for JPA
    constructor() : this("", "", "", "", null, "", "", "", false)
}

fun User.toUserEntity(password: String): UserEntity {
    return UserEntity(
        id = this.id.value,
        username = this.username,
        name = this.name,
        lastname = this.lastname,
        phone = this.phone,
        email = this.email,
        password = password,
        role = this.role.asString(),
        active = this.active
    )
}